addons/l10n_id_efaktur/models/efaktur_document.py

```diff
@@ -0,0 +1,299 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, RedirectWarning
+from odoo.tools import float_repr, float_round
+
+FK_HEAD_LIST = ['FK', 'KD_JENIS_TRANSAKSI', 'FG_PENGGANTI', 'NOMOR_FAKTUR', 'MASA_PAJAK', 'TAHUN_PAJAK', 'TANGGAL_FAKTUR', 'NPWP', 'NAMA', 'ALAMAT_LENGKAP', 'JUMLAH_DPP', 'JUMLAH_PPN', 'JUMLAH_PPNBM', 'ID_KETERANGAN_TAMBAHAN', 'FG_UANG_MUKA', 'UANG_MUKA_DPP', 'UANG_MUKA_PPN', 'UANG_MUKA_PPNBM', 'REFERENSI', 'KODE_DOKUMEN_PENDUKUNG']
+
+LT_HEAD_LIST = ['LT', 'NPWP', 'NAMA', 'JALAN', 'BLOK', 'NOMOR', 'RT', 'RW', 'KECAMATAN', 'KELURAHAN', 'KABUPATEN', 'PROPINSI', 'KODE_POS', 'NOMOR_TELEPON']
+
+OF_HEAD_LIST = ['OF', 'KODE_OBJEK', 'NAMA', 'HARGA_SATUAN', 'JUMLAH_BARANG', 'HARGA_TOTAL', 'DISKON', 'DPP', 'PPN', 'TARIF_PPNBM', 'PPNBM']
+
+
+def _csv_row(data, delimiter=',', quote='"'):
+    return quote + (quote + delimiter + quote).join([str(x).replace(quote, '\\' + quote) for x in data]) + quote + '\n'
+
+
+class EfakturDocument(models.Model):
+    _name = "l10n_id_efaktur.document"
+    _description = "E-faktur Document"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+
+    name = fields.Char(
+        compute='_compute_name',
+        store=True,
+        readonly=False,
+        required=True,
+        precompute=True,
+    )
+    company_id = fields.Many2one('res.company', required=True, readonly=True, default=lambda self: self.env.company)
+    active = fields.Boolean(
+        string="Active",
+        default=True,
+    )
+    invoice_ids = fields.One2many(
+        comodel_name="account.move",
+        inverse_name="l10n_id_efaktur_document",
+        domain="[('move_type', 'in', ['out_invoice', 'out_refund']), ('company_id', '=', company_id), ('l10n_id_efaktur_document', '=', False), ('l10n_id_tax_number', '!=', False), ('state', '=', 'posted')]",
```

Comment:
The documents are not made to be manually edited (at least, not easily. Still possible)
But it should work well for the special cases when the user wants to do that.

I don't really understand your issue though, in the form view if you select an invoice it will be in the list as expected.
Do you mean it disappears from the invoice list view wizard where you select the invoices?

![image](https://github.com/user-attachments/assets/a2431aee-c6d9-4f36-aaf5-d301a675dfdb)
![image](https://github.com/user-attachments/assets/22687c84-9e83-4773-bd71-7c3efe928c9e)
![image](https://github.com/user-attachments/assets/c372358e-1dad-4c58-b8f8-50fcc147b9a9)

Reply:
what I meant is: at the moment you select an invoice in the o2m and saves the record... won't the invoice you selected just disappear from the o2m because of the domain on it filtering it out?

Reply:
I think not. If you look at the screenshots, I had one invoice, then added another one and saved.
The field shows two invoices in the form (sorry if I misunderstand)

I think it works similarly to selecting payments on a batch payment.

Reply:
brain not fully working anymore, I'm afraid... thanks for your patience